app/controllers/search_controller.rb

```diff
@@ -0,0 +1,44 @@
+class SearchController < ApplicationController
+
+  # The organization_display.jsx react component sends an ajax request with the appropriate parameters
+  def search
+    unless params[:term].empty?
+      term = Organization.published.where('name ILIKE ? OR overview ILIKE ?', "%#{params[:term]}%", "%#{params[:term]}%").flatten
+    else
+      term = nil
+    end
+
+    if params[:size] == Organization::DEFAULT
+      size = nil
+    elsif params[:size] == Organization::LESSTHAN25
+      size = Organization.published.where(tech_team_size: 1..25)
+    elsif params[:size] == Organization::LESSTHAN50
+      size = Organization.published.where(tech_team_size: 26..50)
+    elsif params[:size] == Organization::GREATERTHAN50
+      size = Organization.published.where('tech_team_size > ?', 50)
+    else
+      flash[:alert] = "Invalid search!"
+    end
+
+    unless params[:tech].empty?
+      techs = []
+      # This gets the ids of the specified techs
+      techs = params[:tech].split('+').map{|x| x.to_i}
+      org_ids = OrganizationTechnology.where(technology_id: techs).pluck('organization_id').uniq
+      techs = []
+      org_ids.each do |o|
+        techs.concat(Organization.published.where(id: o))
+      end
+      techs.flatten!
+    end
```

Comment:
I refactored to put the logic into the model rather then the controller. I couldn't get the join search to work so its currently still the same logic but no longer in the controller.